### PR TITLE
Use SafeERC20.forceApprove in safeIncreaseAllowance and safeDecreaseAllowance

### PR DESCRIPTION
**Pull request by Amxx**
Fixes https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4258#discussion_r1196774849

#### PR Checklist

- [x] Tests

### CODE REVIEW

**Review by ernestognw — DISMISSED** [review]
LGTM but I think it'll create conflicts with #4258 that will require reapproval. Just let me know

File: test/token/ERC20/utils/SafeERC20.test.js [hunk 1]
```diff
@@ -193,16 +193,19 @@ contract('SafeERC20', function (accounts) {
...
-        await expectRevert(this.mock.$safeIncreaseAllowance(this.token.address, spender, 10), 'USDT approval failure');
+      it('safeIncreaseAllowance works', async function () {
+        this.mock.$safeIncreaseAllowance(this.token.address, spender, 10);
+        expect(this.token.allowance(this.mock.address, spender, 90));
       });
 
-      it('safeApprove can decrease approval', async function () {
-        await expectRevert(this.mock.$safeDecreaseAllowance(this.token.address, spender, 10), 'USDT approval failure');
+      it('safeDecreaseAllowance works', async function () {
+        this.mock.$safeDecreaseAllowance(this.token.address, spender, 10);
```
Comment (frangio) [inline test/token/ERC20/utils/SafeERC20.test.js:202]:
```suggestion
        await this.mock.$safeDecreaseAllowance(this.token.address, spender, 10);
```

File: test/token/ERC20/utils/SafeERC20.test.js [hunk 2]
```diff
@@ -193,16 +193,19 @@ contract('SafeERC20', function (accounts) {
         await this.mock.$safeApprove(this.token.address, spender, 0);
       });
 
-      it('safeApprove can increase approval', async function () {
-        await expectRevert(this.mock.$safeIncreaseAllowance(this.token.address, spender, 10), 'USDT approval failure');
+      it('safeIncreaseAllowance works', async function () {
+        this.mock.$safeIncreaseAllowance(this.token.address, spender, 10);
```
Comment (frangio) [inline test/token/ERC20/utils/SafeERC20.test.js:197]:
```suggestion
        await this.mock.$safeIncreaseAllowance(this.token.address, spender, 10);
```

**Comment by frangio** [comment]
Needs a changeset!

**Review by frangio — APPROVED** [review]

**Review by ernestognw — APPROVED** [review]

**Review by frangio — APPROVED** [review]
LGTM!

**Pull request merged**